### PR TITLE
Cart-widget_CU-knbhrw

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-knbhrw Cart widget](https://app.clickup.com/t/knbhrw)

**Comment by lpboucher** [comment]
Task linked: [CU-knbhrw Cart widget](https://app.clickup.com/t/knbhrw)

File: src/app/core/cart/cart.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    ngOnDestroy() {
+        this.subscriptions.unsubscribe();
+    }
+
+    private evaluateCartWeight(): number {
+        const cartItems = this.snipcart.store.getState().cart.items.items;
+        return cartItems
+            .filter((item: any) => item?.dimensions?.weight != null && item?.quantity != null)
```
Comment (a-wilkes) [inline src/app/core/cart/cart.service.ts:44]: At the moment, this filters everything as none of the items in the cart ever have a non-null weight attribute. I assume this is something that will be fixed when we add a drop-down for picking the weight in the future?

File: src/app/core/cart/cart.service.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    private static readonly REFRESH_RATE: number = 1000;
+
+    private subscriptions: Subscription = new Subscription();
+    private snipcart: any = {};
+    cartTotal$: Observable<number> = new Observable();
+    cartWeight$: Observable<number> = new Observable();
+
+    constructor() {
+        this.subscriptions.add(
+            fromEvent(document, "snipcart.ready")
```
Comment (a-wilkes) [inline src/app/core/cart/cart.service.ts:17]: This looks like the way to subscribe to this event. It definitely works, at least! Do you still want to make a service for this as you mentioned with `EventManager`, or is it better to just do this where we need to?

File: src/app/shared/components/cart-widget/cart-widget.component.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    @apply text-indigo-500;
+
+    @apply md:mb-1 md:mx-auto md:block;
+}
+
+.cart-contents {
+    @apply text-gray-900 inline;
+
+    @apply md:font-medium md:text-xl;
+}
```
Comment (a-wilkes) [inline src/app/shared/components/cart-widget/cart-widget.component.scss:19]: This is now identical to the wallet widget (along with most of the HTML). Should we abstract this down into shared widget component?

File: src/app/core/cart/cart.service.ts (hunk 2 above)
Comment (lpboucher) [reply src/app/core/cart/cart.service.ts:17]: I have implemented a solution, I will explain

File: src/app/core/cart/cart.service.ts (hunk 1 above)
Comment (lpboucher) [reply src/app/core/cart/cart.service.ts:44]: Yes, that's fine for now

File: src/app/shared/components/cart-widget/cart-widget.component.scss (hunk 3 above)
Comment (lpboucher) [reply src/app/shared/components/cart-widget/cart-widget.component.scss:19]: that's a good idea. let's see if we figure out a solution

**Review by lpboucher — APPROVED** [review]

**Pull request merged**